picard/script.py

```diff
@@ -910,6 +910,159 @@ def func_is_video(parser):
         return ""
 
 
+def func_find_str(parser, haystack, needle):
+    """Find the location of the first occurrence of one string within another.
+
+    Arguments:
+        haystack: The string to search
+        needle: The substring to find
+
+    Returns:
+        The zero-based index of the first occurrance of needle in haystack, or -1 if needle was not found.
+    """
+    return str(haystack.find(needle))
+
+
+def func_reverse_str(parser, text):
+    """Returns 'text' in reverse order.
+
+    Arguments:
+        text: Text to be processed
+
+    Returns:
+        Text in reverse order.
+    """
+    return text[::-1]
+
+
+def func_substr(parser, text, start_index, end_index):
+    """Extract a specified portion of a string.
+
+    Arguments:
+        text: The string from which the extract will be made.
+        start_index: Index of the first character to extract.
+        end_index: Index of the first character that will not be extracted.
+
+    Returns:
+        Returns the substring beginning with the character at the start index,
+        up to (but not including) the character at the end index.  The first
+        character is at index number 0.  If the start index is left blank, it
+        defaults to the first character in the string.  If the end index is
+        left blank, it defaults to the number of characters in the string.
+        If either index is negative, it is subtracted from the total number of
+        characters in the string to provide the index used.
+    """
+    try:
+        start = int(start_index) if start_index else None
+    except ValueError:
+        start = None
+    try:
+        end = int(end_index) if end_index else None
+    except ValueError:
+        end = None
+    return text[start:end]
+
+
+def func_get_multi(parser, multi, item_index, separator=MULTI_VALUED_JOINER):
+    """Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+
+    Arguments:
+        multi: The multi-value from which the item is to be retrieved.
+        item_index: The zero-based index of the item to be retrieved.
+
+    Returns:
+        Returns the value of the item at the specified index in the multi-value variable.
+    """
+    if item_index:
+        try:
+            index = int(item_index.eval(parser))
+        except ValueError:
+            return ''
+    else:
+        return ''
+    multi_var = _get_multi_values(parser, multi, separator)
+    try:
+        return str(multi_var[index])
+    except IndexError:
+        return ''
```

Comment:
I think it could be rewritten to a simpler form:

```python
    if not item_index:
        return ''
    try:
        index = int(item_index.eval(parser))
        multi_var = _get_multi_values(parser, multi, separator)
        return str(multi_var[index])
    except (ValueError, IndexError):
        return ''
```

Reply:
Much cleaner!  I'll make the change.